src/sysl/importers/import_swagger.py

```diff
@@ -261,23 +286,54 @@ def info_attrs(info, prefix=''):
                     properties = extract_properties(tspec)
                     if properties:
                         for (fname, fspec) in sorted(properties.iteritems()):
-                            (ftype, fdescr) = self.parse_typespec(fspec)
-                            w('{} <: {}{}',
-                              fname,
-                              ftype if is_sysl_array_type(ftype) or ftype.endswith('*') else ftype + '?',
-                              ' "' + fdescr + '"' if fdescr else '')
+                            (ftype, fdescr) = self.parse_typespec(
+                                fspec, fname, tname)
+                            w('{} <: {}{}{}{}',
+                              fname if fname not in SYSL_TYPES else fname + "_",
+                              ftype if is_sysl_array_type(
+                                  ftype) or ftype.endswith('*') else ftype + '?',
+                              ':' if fdescr or tag else '',
+                              self.getTag(fdescr, "description"), self.getTag(fname, tag))
                     # handle top-level arrays
                     elif tspec.get('type') == 'array':
 
-                        (ftype, fdescr) = self.parse_typespec(tspec)
-                        w('{} <: {}{}',
-                          fname,
-                          ftype if is_sysl_array_type(ftype) or ftype.endswith('*') else ftype + '?',
-                          ' "' + fdescr + '"' if fdescr else '')
+                        (ftype, fdescr) = self.parse_typespec(
+                            tspec, fname, tname)
+                        w('{} <: {}{}{}{}',
+                          fname if fname not in SYSL_TYPES else fname + "_",
+                          ftype if is_sysl_array_type(
+                              ftype) or ftype.endswith('*') else ftype + '?',
+                          ':' if fdescr or tag else '',
+                          self.getTag(fdescr, "description"), self.getTag(fname, tag))
```

Comment:
These two blocks should have been refactored into a function before. They definitely should be now. It can be a local function just above `if properties:`.